sklearn/gaussian_process/gpc.py

```diff
@@ -0,0 +1,707 @@
+        of sqrt(W) is stored.
+    """
+    def __init__(self, kernel=None, jitter=0.0, optimizer="fmin_l_bfgs_b",
+                 n_restarts_optimizer=1, max_iter=100, warm_start=False,
+                 copy_X_train=False, random_state=None):
+        self.kernel = kernel
+        self.jitter = jitter
+        self.optimizer = optimizer
+        self.n_restarts_optimizer = n_restarts_optimizer
+        self.max_iter = max_iter
+        self.warm_start = warm_start
+        self.copy_X_train = copy_X_train
+        self.random_state = random_state
+        self.rng = check_random_state(self.random_state)
+
+    def fit(self, X, y):
+        """Fit Gaussian process classification model
+
+        Parameters
+        ----------
+        X : array-like, shape = (n_samples, n_features)
+            Training data
+
+        y : array-like, shape = (n_samples,)
+            Target values, must be binary
+
+        Returns
+        -------
+        self : returns an instance of self.
+        """
+        if self.kernel is None:  # Use an RBF kernel as default
+            self.kernel_ = \
+                C(1.0, c_bounds="fixed") * RBF(1.0, l_bounds="fixed")
+        else:
+            self.kernel_ = clone(self.kernel)
+
+        self.X_train_ = np.copy(X) if self.copy_X_train else X
+
+        # Encode class labels and check that it is a binary classification
+        # problem
+        label_encoder = LabelEncoder()
+        self.y_train_ = label_encoder.fit_transform(y)
+        self.classes_ = label_encoder.classes_
+        if self.classes_.size > 2:
+            raise ValueError("%s supports only binary classification. "
+                             "y contains classes %s"
+                             % (self.__class__.__name__, self.classes_))
+        elif self.classes_.size == 1:
+            raise ValueError("{0:s} requires 2 classes.".format(
+                self.__class__.__name__))
+
+        if self.kernel_.n_dims == 0:  # no tunable hyperparameters
+            pass
+        elif self.optimizer is not None:
+            # Choose hyperparameters based on maximizing the log-marginal
+            # likelihood (potentially starting from several initial values)
+            def obj_func(theta, eval_gradient=True):
+                if eval_gradient:
+                    lml, grad = self.log_marginal_likelihood(
+                        theta, eval_gradient=True)
+                    return -lml, -grad
+                else:
+                    return -self.log_marginal_likelihood(theta)
+
+            # First optimize starting from theta specified in kernel
+            optima = [(self._constrained_optimization(obj_func,
+                                                      self.kernel_.theta,
+                                                      self.kernel_.bounds))]
+
+            # Additional runs are performed from log-uniform chosen initial
+            # theta
+            if self.n_restarts_optimizer > 1:
```

Comment:
Semantically speaking, shouldn't the test be `> 0` with `0` being the default value?